Task 4/Task4/Actions/FileAction.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Actions
+{
+    public abstract class FileAction
+    {
+        public abstract void Do();
+        public abstract void RollBack();
+        public abstract void RollUp();
```

Comment:
Why is there 3 types of method? I understand rollback and roll forward, but `Do`?

Oh, I got it (by looking at the code that uses it, which isn't very good, some guy could have no way to see it), it's rather `Commit`, `Apply` and `Revert`